src/openMVG/features/sift/SIFT_describer.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+#include <openMVG/config.hpp>
+#include <openMVG/features/sift/SIFT_vlfeat_describer.hpp>
+#if OPENMVG_IS_DEFINED(OPENMVG_HAVE_POPSIFT)
+#include <openMVG/features/sift/SIFT_popSIFT_describer.hpp>
+#endif
+namespace openMVG {
+namespace features {
+
+class SIFT_ImageDescriber : public Image_describer
+{
+public:
+  SIFT_ImageDescriber(const SiftParams& params = SiftParams(), bool isOriented = true)
+    : _params(params)
+  {
+    // TODO: detect if CUDA is available on the computer
```

Comment:
@gregoire-dl: Could you make another PR with this test?
Just need to list the cuda devices and if there no one, we have to use the cpu version.
```
    int numDevices;
    cudaError_t err = cudaGetDeviceCount(&numDevices);
```